client/src/components/Grid/GridList.vue

```diff
@@ -0,0 +1,283 @@
+        <LoadingSpan v-if="loading" />
+        <BAlert v-else-if="!isAvailable" variant="info" show>
+            <span v-if="searchTerm" v-localize>
+                No entries found for: <b>{{ searchTerm }}</b
+                >.
+            </span>
+            <span v-else v-localize> No entries found. </span>
+        </BAlert>
+        <table v-else class="grid-table">
+            <thead>
+                <th
+                    v-for="(fieldEntry, fieldIndex) in config.fields"
+                    :key="fieldIndex"
+                    class="text-nowrap px-2"
+                    :data-description="`grid header ${fieldIndex}`">
+                    <span v-if="config.sortKeys.includes(fieldEntry.key)">
+                        <BLink @click="onSort(fieldEntry.key)">
+                            <span>{{ fieldEntry.title || fieldEntry.key }}</span>
+                            <span v-if="sortBy === fieldEntry.key">
+                                <Icon v-if="sortDesc" icon="caret-up" data-description="grid sort asc" />
+                                <Icon v-else icon="caret-down" data-description="grid sort desc" />
+                            </span>
+                        </BLink>
+                    </span>
+                    <span v-else>{{ fieldEntry.title || fieldEntry.key }}</span>
+                </th>
+            </thead>
+            <tr v-for="(rowData, rowIndex) in gridData" :key="rowIndex" :class="{ 'grid-dark-row': rowIndex % 2 }">
+                <td
+                    v-for="(fieldEntry, fieldIndex) in config.fields"
+                    :key="fieldIndex"
+                    class="px-2 py-3"
+                    :style="{ width: fieldEntry.width }"
+                    :data-description="`grid cell ${rowIndex}-${fieldIndex}`">
+                    <GridOperations
+                        v-if="fieldEntry.type == 'operations'"
+                        :operations="fieldEntry.operations"
+                        :row-data="rowData"
+                        @execute="onOperation($event, rowData)" />
+                    <GridText v-else-if="fieldEntry.type == 'text'" :text="rowData[fieldEntry.key]" />
+                    <GridLink
+                        v-else-if="fieldEntry.type == 'link'"
+                        :text="rowData[fieldEntry.key]"
+                        @click="fieldEntry.handler(rowData, router)" />
+                    <SharingIndicators
+                        v-else-if="fieldEntry.type == 'sharing'"
+                        :object="rowData"
+                        @filter="(filter) => applyFilter(filter, true)" />
+                    <UtcDate v-else-if="fieldEntry.type == 'date'" :date="rowData[fieldEntry.key]" mode="elapsed" />
+                    <StatelessTags
+                        v-else-if="fieldEntry.type == 'tags'"
+                        clickable
+                        :value="rowData[fieldEntry.key]"
+                        :disabled="fieldEntry.disabled"
```

Comment:
please resolve these type errors.

The title/key ones can be solved through a simple function, or by adding `title: undefined;` to one of the union types.

The `.width` type error seems to be a legitimate runtime concern, as there is no type narrowing done to ensure width exists at that point.

The errors in `GridOperations` seems to be due to faulty types. `FieldOperations` has no type property, and `FieldKey` has no operations property, so either this code can never run, or the types are wrong.

The error in `GridLink` can be resolved by either making more specific types (giving link it's own subtype, with handler as a non-optional), or doing a defined check, if this prop really is optional.

The error on `StatelessTags` can be solved in the same manner as the first error.

Reply:
I narrowed `width` to number and fixed the recognition of type guards. Apparently declaring the type only in the prop was not sufficient. Title/Key pairs are now highlighted if missing. I also added `type` to FieldOperations. I think it would be good to limit the types too by using explicit literals in the future but that seems unnecessary at this stage.

Reply:
I do not want to pull new code with errors on it into dev. Not adding all individual type variants is fine, but in that case please add the missing fields to the generic type definition and add type guards to fix the errors.

Reply:
![image](https://github.com/galaxyproject/galaxy/assets/44241786/f90a75c4-f6a1-4bcc-9ecc-cb611a2163f0)

Reply:
Thanks, warnings have been resolved.